### DISCUSSION TITLE
Tomcat Version 9 vs. 10

### DISCUSSION THREAD

**Discussion by abspann** [kitodo/kitodo-production#6134] [Q&A] [answered]
Eine Frage zum Tomcat: Kitodo läuft ja aktuell mit der Tomcat-Version 9, und in Version 10 gab es größere Änderungen (Namespace jakarta statt javax). Daher funktioniert Kitodo nach meinem Verständnis derzeit nicht mit Tomcat 10. Sind hier schon entsprechende Anpassungen in der Pipeline, damit künftige Kitodo-Releases mit aktuelleren Tomcat-Versionen laufen können?

Ich meine, ich hätte dazu schon mal irgendwo (vermutlich hier im Github) einen Kommentar gefunden, aber ich finde ihn leider nicht mehr. Daher meine Frage auf diesem Kanal, ich hoffe es ist der richtige.

Besten Dank und Grüße
Thomas Kögel (Verbundzentrale BVB)

**Comment by BartChris** [comment, marked as answer]
Die Unterstützung von Tomcat 10 ist für den Release 4.0 geplant, befindet sich also noch in der Entwicklung.
https://github.com/kitodo/kitodo-production/milestones

**Comment by stweil** [comment]
Wie ist hier inzwischen der Stand? Im Hauptentwicklungszweig steht in der Datei `pom.xml` inzwischen die Version 10.1.46. Es gibt aber auch schon eine 10.1.52. Gleichzeitig gibt es noch Verweise auf Tomcat 8 (docs/developer/gettingstarted/development-version.md) und Tomcat 8 (Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java).

Aktuelle Linux-Distributionen kommen mit Tomcat 11. Gibt es damit schon Erfahrungen?

**Reply by solth** [reply]
Kompatibilität zu Java 17/21 und Tomcat 10.1 wurde in https://github.com/kitodo/kitodo-production/pull/6692 hergestellt. Diese Änderung wird wie oben beschrieben Teil des nächsten Major Releases 4.0 sein. Bevor dieses veröffentlicht werden kann, müssen jedoch noch ein paar Fehler im Metadaten-Editor und anderen Stellen behoben werden (siehe oben von @BartChris verlinkter Milestone). Mit Tomcat 11 habe ich bisher noch keine Erfahrung gemacht.
